x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/QueryRuleCriteria.java

```diff
@@ -23,84 +26,80 @@
-                    return type;
-                }
-            }
-            throw new IllegalArgumentException("Unknown CriteriaType: " + criteriaType);
-        }
-
-        @Override
-        public String toString() {
-            return name().toLowerCase(Locale.ROOT);
-        }
-    }
+    private static final Logger logger = LogManager.getLogger(QueryRuleCriteria.class);
 
     /**
      *
-     * @param criteriaType The {@link CriteriaType}, indicating how the criteria is matched
-     * @param criteriaMetadata The metadata for this identifier, indicating the criteria key of what is matched against
-     * @param criteriaValue The value to match against when evaluating {@link QueryRuleCriteria} against a {@link QueryRule}
+     * @param criteriaType The {@link QueryRuleCriteriaType}, indicating how the criteria is matched
+     * @param criteriaMetadata The metadata for this identifier, indicating the criteria key of what is matched against.
+     *                         Required unless the CriteriaType is ALWAYS.
+     * @param criteriaValues The values to match against when evaluating {@link QueryRuleCriteria} against a {@link QueryRule}
+     *                      Required unless the CriteriaType is ALWAYS.
      */
-    public QueryRuleCriteria(CriteriaType criteriaType, String criteriaMetadata, Object criteriaValue) {
+    public QueryRuleCriteria(QueryRuleCriteriaType criteriaType, @Nullable String criteriaMetadata, @Nullable List<Object> criteriaValues) {
 
         Objects.requireNonNull(criteriaType);
-        Objects.requireNonNull(criteriaMetadata);
-        Objects.requireNonNull(criteriaValue);
-
-        if ((criteriaType == CriteriaType.EXACT) == false) {
-            throw new IllegalArgumentException("Invalid criteriaType " + criteriaType);
-        }
 
-        if (Strings.isNullOrEmpty(criteriaMetadata)) {
-            throw new IllegalArgumentException("criteriaMetadata cannot be blank");
+        if (criteriaType != ALWAYS) {
+            if (Strings.isNullOrEmpty(criteriaMetadata)) {
+                throw new IllegalArgumentException("criteriaMetadata cannot be blank");
+            }
+            if (criteriaValues == null || criteriaValues.isEmpty()) {
+                throw new IllegalArgumentException("criteriaValues cannot be null or empty");
+            }
         }
 
-        this.criteriaType = criteriaType;
         this.criteriaMetadata = criteriaMetadata;
-        this.criteriaValue = criteriaValue;
+        this.criteriaValues = criteriaValues;
+        this.criteriaType = criteriaType;
+
     }
 
     public QueryRuleCriteria(StreamInput in) throws IOException {
-        this.criteriaType = in.readEnum(CriteriaType.class);
-        this.criteriaMetadata = in.readString();
-        this.criteriaValue = in.readGenericValue();
+        this.criteriaType = in.readEnum(QueryRuleCriteriaType.class);
+        this.criteriaMetadata = in.readOptionalString();
```

Comment:
Serialization changed here and in `writeTo` method. I'd say we're OK with this as this code has not been released yet and serverless is still not on, but the safest would be to use `TransportVersion` to check.

Reply:
Thanks for the confidence check @carlosdelest - this has been addressed in [a4971d5](https://github.com/elastic/elasticsearch/pull/97730/commits/a4971d54b795c91b85203b6e5500eeac4f8533d6)